include/godot_cpp/core/defs.hpp

```diff
@@ -92,6 +92,23 @@
 #define unlikely(x) x
 #endif
 
+#ifdef REAL_T_IS_DOUBLE
+typedef double real_t;
```

Comment:
Unavoidable to re-introduce this. @reduz said something about when methods are called over the GD extension bridge doubles are always used even if members are defined as a float (where we do have a bug I think because it failed for me), in this case however we're re-implementing everything on our side so we need the right typing to be binary compatible.